### ISSUE TITLE
escaping textarea and text values

### ISSUE THREAD

**Issue by mainpart** [Codestar/codestar-framework#115]
When sanitizing value the 'sanitize' option stands for escaping data, that further has to be written to the database. If i'm using textarea/code_editor field to store html that includes <textarea></textarea> field everything goes wrong because of unescaped </textarea> tag. The code for code_editor textarea field is pretty straightforward:

```php
      echo $this->field_before();
      echo '<textarea name="'. esc_attr( $this->field_name() ) .'"'. $this->field_attributes() .' data-editor="'. esc_attr( json_encode( $settings ) ) .'">'. $this->value .'</textarea>';
      echo $this->field_after();
```

I beleive it need esc_textarea($this->value) or at least 
```php
apply_filter('csf_'.$this->field['type'].'_value' . $this->value);
```
to be able to display quotes, opening and closes tags code correctly

**Comment by Codestar** [comment]
Hi, I didn't get the question. Shortly, you can disable standard sanitize with this parameter:

```php
sanitize => false,
```
or you can set your custom sanitize like this. ( [more information](https://codestarframework.com/documentation/#/faq?id=how-to-use-sanitize-) )

```php
sanitize => 'your_function_callback',
```

**Comment by mainpart** [comment]
Hello again. Yep, i read about sanitize and experimented with it quite well. Sanitize callback is about writing sanitized version of input to database. For example if you don't want to allow any html markup you assign `strip_tags` as sanitizing function and the job done - any user input will be stored without tags. Next time user opens the editor - viola, all text will be without markup. 

But what if user want's to store html template? Tags can be freely stored inside mysql (all escaping upon save is made by wordpress core) but next time user  reloads the page, he will get unescaped markup inside ```<input value="">```  or inside ```<textarea>...</textarea>``` tags. Any quotes (in first case) or closing textarea tag will broke page. I've faced this issue trying to save html form markup inside codestar textarea field. 

I've tried to use htmlentities as sanitizing function  but that was wrong - my html template was stored as escaped markup inside database ( there were &lt; and &gt; and other entities ) and each time i pulled it on the front side of site, i had to remember to do htmlentities_decode which was completely unnecessary and ideologicaly wrong.

So my idea that any value should be escaped when outputed inside input or textarea tag. I'm using ```sanitize=>false``` to save exact value to database and this value needs to be escaped when outputed inside admin part in tags attributes or tags values.   

As far as i seen you already use attr_esc core wordpres function in some components. Why it's missing from described above?

**Comment by mainpart** [comment]
TLDR:
you'll break markup trying to save </textarea> text inside code editor without sanitize:

```php
	CSF::createSection( $prefix, array(
		'fields' => array(
			array(
				'type'  => 'code_editor',
				),
```
and in case of sanitizing ,you will save sanitized text in db. You will need to de-sanitize each time you get it with get_option in any other part of site (what i beleive is wrong approach). 

Basic idea - you can save text with any sanitization or can save without it, but you need to escape that text upon output inside textarea tag in admin side.

**Comment by mainpart** [comment]
does it make sense?
